hail/python/hail/table.py

```diff
@@ -2310,17 +2310,22 @@ def join(self,
 
         Notes
         -----
-        Hail supports four types of joins specified by `how`:
-
-        - **inner** -- Key must be present in both the left and right tables.
-        - **outer** -- Key present in either the left or the right. For keys
-          only in the left table, the right table's fields will be missing.
-          For keys only in the right table, the left table's fields will be
-          missing.
-        - **left** -- Key present in the left table. For keys not found on
-          the right, the right table's fields will be missing.
-        - **right** -- Key present in the right table. For keys not found on
-          the right, the right table's fields will be missing.
+        Tables are joined at rows whose key fields have equal values. Missing values never match.
```

Comment:
Moved the missing values note here. 2 sentences I think easier to read, because non-missing key can mean "keys in one table and not the other" (which is expanded on below, illegal) or "key values that are NA/missing"